Array.cc

```diff
@@ -1315,6 +1315,51 @@ bool Array::check_semantics(string &msg, bool)
     return sem;
 }
 
+
+/**
+ * Makes the square bracket representation, with dimension names where available, of
+ * the array's dimensions, ex: [5][lat=100][lon=200]
+ * @param a The array to examine
+ * @return The square bracket representation of the array's dimensions
+ */
+string get_dims_decl(Array &a) {
```

Comment:
Make the Array const.

Reply:
I cannot make it const because I am using `Array::dim_begin()` and `Array::dim_end()` like this:

    for(auto itr=a.dim_begin(); itr!=a.dim_end(); itr++){

And since neither `Array::dim_begin()` or `Array::dim_end()` are **not** marked const there is an issue.